clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -412,3 +394,53 @@ func GetNumObjectsMatchingSelector(c dynamic.Interface, namespace string, resour
 	err := client.RetryWithExponentialBackOff(client.RetryFunction(listFunc))
 	return numObjects, err
 }
+
+// The pod can only schedule onto nodes that satisfy requirements in both NodeAffinity and nodeSelector.
+func podMatchesNodeLabels(podSpec corev1.PodSpec, node corev1.Node) bool {
+	// Check if node.Labels match pod.Spec.NodeSelector.
+	if len(podSpec.NodeSelector) > 0 {
+		selector := labels.SelectorFromSet(podSpec.NodeSelector)
+		if !selector.Matches(labels.Set(node.Labels)) {
+			return false
+		}
+	}
+
+	// 1. nil NodeSelector matches all nodes (i.e. does not filter out any nodes)
+	// 2. nil []NodeSelectorTerm (equivalent to non-nil empty NodeSelector) matches no nodes
+	// 3. zero-length non-nil []NodeSelectorTerm matches no nodes also, just for simplicity
+	// 4. nil []NodeSelectorRequirement (equivalent to non-nil empty NodeSelectorTerm) matches no nodes
+	// 5. zero-length non-nil []NodeSelectorRequirement matches no nodes also, just for simplicity
+	// 6. non-nil empty NodeSelectorRequirement is not allowed
+	nodeAffinityMatches := true
+	affinity := podSpec.Affinity
+	if affinity != nil && affinity.NodeAffinity != nil {
+		nodeAffinity := affinity.NodeAffinity
+		// if no required NodeAffinity requirements, will do no-op, means select all nodes.
+		if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+			return true
+		}
+
+		// Match node selector for requiredDuringSchedulingIgnoredDuringExecution.
+		if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
```

Comment:
Isn't this check already done above?

Reply:
all have done except `RequiredDuringSchedulingRequiredDuringExecution`, since it has not implemented

Reply:
I think that Krzysiek's point was that there is no need to check for != nil here. nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution has to bee not nil here as otherwise you would exit from the fun in line 416

Reply:
> I think that Krzysiek's point was that there is no need to check for != nil here.

That's right. :+1:

Reply:
This still isn't resolved - please remove this check - after if from line 452, we already know it's not nil.